packages/paste-core/components/toast/src/useToaster.ts

```diff
@@ -3,21 +3,51 @@ import {uid} from '@twilio-paste/uid-library';
 import type {ToasterToast, ToasterPush, UseToasterReturnedProps} from './types';
 
 export const useToaster = (): UseToasterReturnedProps => {
+  const isMounted = React.useRef<boolean | null>(null);
   const [toasts, setToasts] = React.useState<ToasterToast[]>([]);
 
+  React.useEffect(() => {
+    isMounted.current = true;
+    return () => {
+      isMounted.current = false;
+      /**
+       * Clear all timeouts from the toaster when the component unmounts
+       */
+      toasts.forEach((toast) => {
+        if (toast.timeOutId) {
+          window.clearTimeout(toast.timeOutId);
+        }
+      });
+    };
+  }, []);
```

Comment:
Mmmmm so this is the small potential memory leak. If you trash the DOM, navigate away, there can be timeouts still running.

So we clean up on unmount in this effect.

Reply:
Correct